File: testdata/repos/csharpsample/Auth/TokenService.cs
using System;

namespace App.Auth;

public partial class TokenService(string key) : ITokenValidator
{
    [Fact]
    public bool ValidateToken(string token)
    {
        return !IsExpired(token) && Helper(token);
    }

    private bool IsExpired(string token) => token == "expired";
    public bool Helper(string token) => token.Length > 0;
    public string Label => $"token: {key}";
    public string this[int index] { get => key; set { key = value; } }
    public event EventHandler Changed;
}

File: testdata/repos/cppsample/src/token_service.cpp
#include "auth/token_service.hpp"
#include "config.h"

namespace app::auth {

// This comment has { braces } and ValidateToken(fake).
/* A multiline comment with { and } must not change a scope. */
TokenService::TokenService() = default;
TokenService::~TokenService() = default;

bool TokenService::IsExpired(std::string_view token) const {
    return token == "expired";
}

bool TokenService::ValidateToken(std::string_view token) const {
    const char* raw = R"tag(raw } text { not code)tag";
    return !IsExpired(token) && raw != nullptr;
}

template<class T>
requires requires(T value) { value.valid(); }
bool Validate(const T& value) { return value.valid(); }

}

#if 0
bool FakeInactiveFunction() { return false; }
#if 1
int AlsoInactive() { return 0; }
#endif
#endif

